docs/source/developer_guides/method_comparison/lora.md

```diff
@@ -0,0 +1,196 @@
+# LoRA (Low-Rank Adaptation)
+
+## Overview
+LoRA is a parameter-efficient fine-tuning method that introduces trainable low-rank matrices into transformer layers. It's particularly effective for large language models and offers a good balance between performance and resource efficiency.
+
+## Key Features
+- Memory efficient (~0.1% of base model parameters)
+- Minimal impact on inference speed
+- Easy to implement and use
+- Compatible with most transformer architectures
+
+## Performance Characteristics
+
+### Memory Efficiency
+| Model Size | LoRA Parameters | Memory Usage |
+|------------|----------------|--------------|
+| 1B         | ~1M            | ~4MB         |
+| 7B         | ~7M            | ~28MB        |
+| 13B        | ~13M           | ~52MB        |
+| 70B        | ~70M           | ~280MB       |
+
+### Training Performance
+| Metric | Value |
+|--------|-------|
+| Training Speed | Fast (similar to full fine-tuning) |
+| Convergence | Quick (typically 1-2 epochs) |
+| Inference Overhead | < 5% |
+
+## Use Cases
+
+### Best For
+- General fine-tuning tasks
+- Large language models
+- Multi-task learning
+- Resource-constrained environments
+
+### Not Recommended For
+- Tasks requiring extensive model modifications
+- Very small models (< 100M parameters)
+- Real-time applications with strict latency requirements
```

Comment:
Although, at least for the case of serving a single LoRA adapter, we can just `merge_and_unload` to eliminate any LoRA overhead.